hooks/tk-multi-publish2/basic/publish_distortion_maps.py

```diff
@@ -0,0 +1,340 @@
+
+            {
+                "Settings Name": {
+                    "type": "settings_type",
+                    "default": "default_value",
+                    "description": "One line description of the setting"
+            }
+
+        The type string should be one of the data types that toolkit accepts as
+        part of its environment configuration.
+        """
+        # inherit the settings from the base publish plugin
+        base_settings = super(SyntheyesDistortionMapsPublishPlugin, self).settings or {}
+
+        # settings specific to this class
+        syntheyes_publish_settings = {
+            "Publish Template": {
+                "type": "template",
+                "default": None,
+                "description": "Template path for published work files. Should"
+                               "correspond to a template defined in "
+                               "templates.yml.",
+            },
+            "Undistort Template": {
+                "type": "template",
+                "default": None,
+                "description": "String template for publishing undistort images. Should"
+                               "correspond to a string template defined in templates.yml.",
+            },
+            "Redistort Template": {
+                "type": "template",
+                "default": None,
+                "description": "String template for publishing redistort images. Should"
+                               "correspond to a string template defined in templates.yml.",
+            },
+        }
+
+        # update the base settings
+        base_settings.update(syntheyes_publish_settings)
+
+        return base_settings
+
+    @property
+    def item_filters(self):
+        """
+        List of item types that this plugin is interested in.
+
+        Only items matching entries in this list will be presented to the
+        accept() method. Strings can contain glob patters such as *, for example
+        ["maya.*", "file.maya"]
+        """
+        return ["syntheyes.distortion_maps"]
+
+    @property
+    def item_property_cache(self):
+        """
+        Backup of each valid item's properties. Since, in this case, mutliple
+        plugins may work on the same item with varying publish paths, it cannot 
+        be guaranteed that the properties of the item will remain the same as 
+        they were at the time of validation.
+        Therefore, the state of the item's properties is cached here and re-set
+        before publishing.
+        """
+        if not hasattr(self, "_item_property_cache"):
+            self._item_property_cache = {}
+        return self._item_property_cache
+
+    def accept(self, settings, item):
+        """
+        Method called by the publisher to determine if an item is of any
+        interest to this plugin. Only items matching the filters defined via the
+        item_filters property will be presented to this method.
```

Comment:
item_filters ist aber nur definiert als ["syntheyes.distortion_maps"]. Du hast im Code aber auch noch redistort und undistort stehen. Fehlt da noch was, oder hat das nichts mit einander zu tun?

Reply:
Das System von Collector & Publish Plugin funktioniert so, dass jedes Item, das vom Collector aus der Szene gesammelt wurde, an jedes registrierte Plugin geht. Um die Liste schnell filtern zu können wird zuerst die item_filters-Property angewendet: Nur items, die einem der Filter entsprechen, gehen dann weiter an accept(). Der Collector baut daher für jede Camera in SynthEyes mit Distortion-Informationen ein Item vom Type "syntheyes.distortion_maps", das dann hier nur an das DistortionMaps-PublishPlugin geht.
Undistort und Redistort sind für die jeweilige Bennenung der beide Maps selbst.